deeplake/core/chunk/base_chunk.py

```diff
@@ -570,6 +570,11 @@ def is_empty_tensor(self):
         return len(self.tensor_meta.max_shape) == 0 and len(self.data_bytes) == 0
 
     def _text_sample_to_byte_string(self, sample):
+        if isinstance(sample, deeplake.Tensor):
+            try:
+                return sample.numpy(aslist=True)[0].encode("utf-8")
```

Comment:
why not 
sample = sample.numpy(aslist=True)[0]
the encoding and error catching will be done further down the method